src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/_params.py

```diff
@@ -215,8 +215,13 @@ def load_arguments(self, _):
 
     with self.argument_context('aks update-credentials') as c:
         c.argument('reset_service_principal', action='store_true')
+        c.argument('reset_aad', action='store_true')
         c.argument('service_principal')
         c.argument('client_secret')
+        c.argument('aad_client_app_id')
+        c.argument('aad_server_app_id')
+        c.argument('aad_server_app_secret')
+        c.argument('aad_tenant_id')
```

Comment:
Lines 219-224 shouldn't be needed. The CLI will already pick them up and these aren't adding any metadata.

Reply:
However, I would recommend splitting the these into two argument groups like so:
```Python
with self.argument_context('aks update-credentials', arg_group='Service Principal') as c: 
        c.argument('reset_service_principal', action='store_true') 
        c.argument('service_principal') 
        c.argument('client_secret') 

with self.argument_context('aks update-credentials', arg_group='AAD') as c: 
        c.argument('reset_aad', action='store_true') 
        c.argument('aad_client_app_id') 
        c.argument('aad_server_app_id') 
        c.argument('aad_server_app_secret') 
        c.argument('aad_tenant_id')
```

Reply:
Thanks for the detailed feedback, much appreciated. I have implemented the change but could you explain why lines 219-224 would be picked up by the CLI but 217-218 wouldn't be (hoping to avoid this mistake for next time).

Reply:
The two properties that have `action='store_true'` WOULD be picked up by the CLI, but they would not behave as flags. They would expect a value. Hence you need those two `c.argument(…)` entries to alter their default behavior. However, based on my recommendation to group the arguments in help, you actually do need to specify them all.

Reply:
Ok makes sense - thanks